### PR TITLE
Feat/report update endpoint

### PR DESCRIPTION
**Pull request by fahmirzldi**
create report update endpoint

### CODE REVIEW

File: app/main/model/report.py [hunk 1]
```diff
@@ -78,7 +79,7 @@ def create_report(self, data):
         except Exception as e:
             logging.exception("An error occurred while creating a report: %s", str(e))
             return None
-    
+
     def get_all_reports(self):
```
Comment (fatjan) [inline app/main/model/report.py:83]: do you want to implement the pagination here or will that be in the new PR?

File: tests/service/test_report_service.py [hunk 2]
```diff
@@ -108,3 +130,25 @@ def test_get_an_report(self, mock_get_report_by_id):
         self.assertEqual(result["public_id"], data["public_id"])
         self.assertEqual(result["reason"], data["reason"])
         self.assertEqual(result["item_id"], data["item_id"])
+
+    @patch("app.main.model.report.Report.update_report")
+    def test_update_report(self, mock_update_report):
+        # ARRANGE
+        data = report_data_1
+        data["status"] = "accepted"
```
Comment (fatjan) [inline tests/service/test_report_service.py:138]: I thought, the status is boolean, no? in the dto

File: app/main/model/report.py (hunk 1 above)
Comment (fahmirzldi) [reply app/main/model/report.py:83]: Will be done on new PR

File: tests/service/test_report_service.py (hunk 2 above)
Comment (fahmirzldi) [reply tests/service/test_report_service.py:138]: I think the API contract is wrong, as mentioned here #15, the status is 

- Received
- Accepted
- Rejected

File: app/main/util/dto.py [hunk 3]
```diff
@@ -88,4 +88,5 @@ class ReportDto:
             "item_id": fields.String(required=True, description="report item_id"),
             "reason": fields.String(required=True, description="report reason"),
         },
-    )
\ No newline at end of file
+    )
+    status = api.model("status", {"status": fields.Boolean(description="status")})
```
Comment (fatjan) [inline app/main/util/dto.py:92]: I think you need to update the data type on the status to `String` instead @fahmirzldi

File: tests/service/test_report_service.py (hunk 2 above)
Comment (fatjan) [reply tests/service/test_report_service.py:138]: You need to update the dto then. I wrote the comment on it just now.

**Review by fatjan — CHANGES_REQUESTED** [review]
Just a small adjustment needed, as per earlier comment.

File: app/main/util/dto.py (hunk 3 above)
Comment (fahmirzldi) [reply app/main/util/dto.py]: Done please kindly review @fatjan @AchmadQiz

File: tests/controller/test_report_controller.py [hunk 4]
```diff
@@ -25,16 +25,31 @@
...
+    "public_id": str(uuid.uuid4()),
+    "username": "test_user",
+    "email": "@gmail.com",
+    "password": "test_password",
+    "role": "admin",
+    "status": "active",
+}
+
 report_data = {
+    "public_id": str(uuid.uuid4()),
```
Comment (Achmadqizwini) [inline tests/controller/test_report_controller.py:39]: this public_id key is duplicated

**Review by fatjan — APPROVED** [review]

**Pull request merged**